packages/theme/src/cli/services/local-storage.ts

```diff
@@ -54,58 +55,85 @@ function themeStorePasswordStorage() {
 }
 
-export function removeDevelopmentTheme(): void {
+export function removeDevelopmentTheme(
+  themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage(),
+): void {
   outputDebug(outputContent`Removing development theme...`)
-  developmentThemeLocalStorage().delete(getThemeStore())
+  developmentThemeLocalStorage().delete(requireThemeStore(themeStorage))
 }
 
-export function getREPLTheme(): string | undefined {
+export function getREPLTheme(
+  themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage(),
+): string | undefined {
   outputDebug(outputContent`Getting REPL theme...`)
-  return replThemeLocalStorage().get(getThemeStore())
+  return replThemeLocalStorage().get(requireThemeStore(themeStorage))
 }
 
-export function setREPLTheme(theme: string): void {
+export function setREPLTheme(
+  theme: string,
+  themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage(),
+): void {
   outputDebug(outputContent`Setting REPL theme to ${theme}...`)
-  replThemeLocalStorage().set(getThemeStore(), theme)
+  replThemeLocalStorage().set(requireThemeStore(themeStorage), theme)
 }
 
-export function removeREPLTheme(): void {
+export function removeREPLTheme(themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage()): void {
   outputDebug(outputContent`Removing REPL theme...`)
-  replThemeLocalStorage().delete(getThemeStore())
+  replThemeLocalStorage().delete(requireThemeStore(themeStorage))
 }
 
-export function getStorefrontPassword(): string | undefined {
-  const themeStore = getThemeStore()
+export function getStorefrontPassword(
+  themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage(),
+): string | undefined {
+  const themeStore = requireThemeStore(themeStorage)
   outputDebug(outputContent`Getting storefront password for shop ${themeStore}...`)
-  return themeStorePasswordStorage().get(getThemeStore())
+  return themeStorePasswordStorage().get(themeStore)
 }
 
-export function setStorefrontPassword(password: string): void {
-  const themeStore = getThemeStore()
+export function setStorefrontPassword(
+  password: string,
+  themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage(),
+): void {
+  const themeStore = requireThemeStore(themeStorage)
   outputDebug(outputContent`Setting storefront password for shop ${themeStore}...`)
   themeStorePasswordStorage().set(themeStore, password)
 }
 
-export function removeStorefrontPassword(): void {
-  const themeStore = getThemeStore()
+export function removeStorefrontPassword(
+  themeStorage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage(),
+): void {
+  const themeStore = requireThemeStore(themeStorage)
   outputDebug(outputContent`Removing storefront password for ${themeStore}...`)
   themeStorePasswordStorage().delete(themeStore)
 }
+
+function requireThemeStore(storage: LocalStorage<ThemeLocalStorageSchema> = themeLocalStorage()): string {
```

Comment:
Personally, I believe we could rename it to `ensureThemeStore` as we have this `ensure*` pattern in some places :)

Reply:
🤔 
Since this check is a tripwire rather than a `set if not already set` check, I'm going to rename this to `assertThemeStoreExists`.

This avoids conflicting with `ensureThemeStore` which sets the theme store if not set in the command layer [here](https://github.com/Shopify/cli/blob/d549077b51cd5ab646ce7aa0430611e2c3f5eb28/packages/theme/src/cli/utilities/theme-store.ts#L6)